neqo-transport/src/connection/mod.rs

```diff
@@ -1125,7 +1133,13 @@ impl Connection {
         }
         // At this point, we should only have the connection ID that we generated.
         // Update to the one that the server prefers.
-        let path = self.paths.primary();
+        let Some(path) = self.paths.primary() else {
+            self.stats
+                .borrow_mut()
+                .pkt_dropped("Retry without an existing path");
+            return;
```

Comment:
Got a test for this?

Reply:
I'm not sure I can come up with one. `new_client()` does `setup_handshake_path()`, which calls `make_permanent()` on that first path, which should always push it into index 0, i.e., make it primary.

Reply:
Fair enough.  I thought I'd ask though.